laser_encoders/laser_tokenizer.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/python3
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# LASER  Language-Agnostic SEntence Representations
+# is a toolkit to calculate multilingual sentence embeddings
+# and to use them for document classification, bitext filtering
+# and mining
+#
+# --------------------------------------------------------
+#
+# Helper functions for tokenization
+
+import os
+import sys
+import logging
+from pathlib import Path
+import sentencepiece as spm
+from sacremoses import MosesPunctNormalizer, MosesDetokenizer
+
+logging.basicConfig(
+    stream=sys.stdout,
+    level=logging.INFO,
+    format="%(asctime)s | %(levelname)s | %(name)s | %(message)s",
+)
+logger = logging.getLogger("preprocess")
+
+
+class LaserTokenizer:
+    def __init__(
+        self,
+        spm_model: Path,
+        lang: str = "en",
+        lower_case: bool = True,
+        descape: bool = False,
+        verbose: bool = False,
+        over_write: bool = False,
+    ):
+        self.spm_model = spm_model
```

Comment:
It would be nice to also add an assertion here to check if the file exists: `assert spm_model.exists(), f"spm model file: {spm_model} does not exist"`